### PR TITLE
[SRVCOM-2062] Add probe configuration info for Knative deployments

### PR DESCRIPTION
**Pull request by rh-max**
Version(s):
4.8+

Issue:
https://issues.redhat.com/browse/SRVCOM-2062

Link to docs preview:
https://53270--docspreview.netlify.app/openshift-enterprise/latest/serverless/admin_guide/serverless-configuration.html#serverless-configuration-CR-system-deployments

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by rh-max** [comment]
@skonto Hi, could you PTAL? There's a FIXME in the Eventing part, please suggest a fitting example.

**Comment by skonto** [comment]
LGTM

File: modules/knative-eventing-CR-system-deployments.adoc [hunk 1]
```diff
@@ -31,6 +32,10 @@ metadata:
   namespace: knative-eventing
 spec:
   deployments:
+  - name: FIXME
```
Comment (skonto) [inline modules/knative-eventing-CR-system-deployments.adoc:35]: You might want to use `eventing-controller` here. How about the Kafka CR?
cc @pierDipi if he wants a better example.

File: modules/knative-eventing-CR-system-deployments.adoc [hunk 2]
```diff
@@ -31,6 +32,10 @@ metadata:
   namespace: knative-eventing
 spec:
   deployments:
+  - name: eventing-controller
```
Comment (mgencur) [inline modules/knative-eventing-CR-system-deployments.adoc:35]: There's already `- name: eventing-controller` a few lines below so I think we should just add the `readinessProbes` configuration there. The rest of this PR looks ok.

File: modules/knative-eventing-CR-system-deployments.adoc (hunk 1 above)
Comment (rh-max) [reply modules/knative-eventing-CR-system-deployments.adoc]: Opened an issue for Kafka https://issues.redhat.com/browse/SRVCOM-2206

File: modules/knative-eventing-CR-system-deployments.adoc [hunk 3]
```diff
@@ -48,6 +52,7 @@ spec:
     nodeSelector:
       disktype: hdd
 ----
+<1> The `readiness` and `liveness` probe overrides allow overriding all fields of a probe in a container of a deployment as specified in the Kubernetes API except for the fields related to the probe handler: `exec`, `grpc`, `httpGet`, and `tcpSocket`.
```
Comment (maxwelldb) [inline modules/knative-eventing-CR-system-deployments.adoc:55]: ⚠️ ISG: 

> Avoid stating that inanimate objects grant abilities to people, as in "the product allows you to...". Whenever possible, use a direct, user-focused alternative such as "you can use the product to..." or "with this product, users can..."

**Review by maxwelldb — APPROVED** [review]

File: modules/knative-eventing-CR-system-deployments.adoc [hunk 4]
```diff
@@ -6,7 +6,7 @@
 [id="knative-eventing-CR-system-deployments_{context}"]
 = Overriding Knative Eventing system deployment configurations
 
-You can override the default configurations for some specific deployments by modifying the `deployments` spec in the `KnativeEventing` custom resource (CR). Currently, overriding default configuration settings is supported for the `eventing-controller`, `eventing-webhook`, and `imc-controller` fields.
+You can override the default configurations for some specific deployments by modifying the `deployments` spec in the `KnativeEventing` custom resource (CR). Currently, overriding default configuration settings is supported for the `eventing-controller`, `eventing-webhook`, and `imc-controller` fields, as well as for the `readiness` and `liveness` fields for probes.
```
[the remark this answers is not in the document]
Comment (rh-max) [reply modules/knative-eventing-CR-system-deployments.adoc:9]: It helps to have `readiness` and `liveness` fields group together and separate from the rest, so keeping it a sentence helps here.

**Review by bergerhoffer — COMMENTED** [review]
LGTM for merging, just wanted to check first if it has been QE approved?

**Pull request merged**